src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/AddressDtoValidatorTest.java

```diff
@@ -0,0 +1,225 @@
+package uk.gov.companieshouse.overseasentitiesapi.validation;
+
+import org.apache.commons.lang.StringUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.overseasentitiesapi.mocks.AddressMock;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.EntityDto;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationMessages;
+import uk.gov.companieshouse.service.rest.err.Err;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(MockitoExtension.class)
+public class AddressDtoValidatorTest {
+
+    private static final String CONTEXT = "12345";
+
+    private AddressDtoValidator addressDtoValidator;
+    private AddressDto addressDto;
+
+    @BeforeEach
+    public void init() {
+        addressDtoValidator = new AddressDtoValidator();
+        addressDto = AddressMock.getAddressDto();
+    }
+
+    @Test
+    void testValidatePropertyNameNumberIsEmpty(){
+        addressDto.setPropertyNameNumber(" ");
+        String field = EntityDto.PRINCIPAL_ADDRESS_FIELD;
+        Errors errors = addressDtoValidator.validate(field, addressDto, new Errors(), CONTEXT);
+
+        assertError(field, AddressDto.PROPERTY_NAME_NUMBER_FIELD, ValidationMessages.NOT_EMPTY_ERROR_MESSAGE, errors);
+    }
+
+    @Test
+    void testValidatePropertyNameNumberIsNull(){
```

Comment:
And `checkErrorReportedWhenPropertyNameNumberFieldIsNull()`

Reply:
As above.